src/common/functions.py

```diff
@@ -91,46 +90,47 @@ def is_sql_accessible(db_conn: str, query: str) -> bool:
         password = ""
         dsn = ""
 
-        if not db_conn:
-            return False
-        try:
-            user_part, dsn = db_conn.split("@")
-            username, password = user_part.split("/")
-        except ValueError:
-            # If the string does not have the expected format, return False
-            return False
-
-        connection = oracledb.connect(user=username, password=password, dsn=dsn)
-
-        cursor = connection.cursor()
-        if not query:
-            return False
-
-        cursor.execute(query)
-        rows = cursor.fetchmany(2)
-
-        if not rows:
-            logger.error("SQL source return an empty table!")
-            return False
-        desc = cursor.description
-        if len(desc) != 1:
-            logger.error("SQL source returns %s columns, expected 1.",len(desc))
-            return False
-
-        col_type = desc[0][1]
-        if col_type not in (oracledb.DB_TYPE_VARCHAR, oracledb.DB_TYPE_NVARCHAR):
-            # to be implemented: oracledb.DB_TYPE_BLOB, oracledb.DB_TYPE_CLOB, oracledb.DB_TYPE_NCLOB
-            logger.error("SQL source returns column of type %s , expected VARCHAR or BLOB.",col_type
-            )
-            return False
+        ok = True
+
+        if db_conn and query:
+            try:
+                user_part, dsn = db_conn.split("@")
+                username, password = user_part.split("/")
+            except ValueError:
+                # If the string does not have the expected format, return False
+                return False
+
+            connection = oracledb.connect(user=username, password=password, dsn=dsn)
+
+            cursor = connection.cursor()
+
+            cursor.execute(query)
+            rows = cursor.fetchmany(2)
+            desc = cursor.description
+            if not rows:
+                logger.error("SQL source return an empty table!")
+                ok = False
+            if len(desc) != 1:
+                logger.error("SQL source returns %s columns, expected 1.", len(desc))
+                ok = False
+
+            if rows and len(desc) != 1:
+                col_type = desc[0][1]
```

Comment:
Use the FetchInfo attribute for readability? https://python-oracledb.readthedocs.io/en/latest/api_manual/fetch_info.html